src/XR/features/WebXRControllerPointerSelection.ts

```diff
@@ -366,14 +422,63 @@ export class WebXRControllerPointerSelection extends WebXRAbstractFeature {
                     scene.pointerY = this._screenCoordinatesRef.y;
                 }
             }
-            controllerData.pick = this._scene.pickWithRay(controllerData.tmpRay, this._scene.pointerMovePredicate || this.raySelectionPredicate);
 
-            const pick = controllerData.pick;
+            let pick = null;
+            let hoverAtOrigin = false;
+            let hoverRange = false; 
+            if (controllerData.hoverIndexMeshTip) {
+                let hoverPickInfo = this._pickWithMesh(controllerData.hoverIndexMeshTip, false, this.nearPickPredicate);
+                let hoverGrabInfo = this._pickWithMesh(controllerData.hoverIndexMeshTip, false, this.nearGrabPredicate);
+                if ((hoverPickInfo && hoverPickInfo.hit) || (hoverGrabInfo && hoverGrabInfo.hit)) {
+                    hoverRange = true;
+                } 
+                if (hoverRange) {
+                    controllerData.laserPointer.isVisible = false;

```

Comment:
So, that was the initial plan - to populate pick info from hover during grab action. But, this is creating more confusion since we just use "pick" for both hover and grab. It will anyways be overwritten by neargrab pick info when it is successful. Although when it is not successful, it is creating problems with pointermove in attachtrackedRayMode.  I will have to still investigate more on this. but yeah if this could be done, that will be neat!